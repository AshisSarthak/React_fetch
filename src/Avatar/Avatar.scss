.avatar{
    width: 40px;
    height: 40px;
    background: #ddd;
    border-radius: 50%;
    border: 1px solid #eee;
    cursor: pointer;
    display: flex;
    justify-content: center;
    align-items: center;
    position: relative;
    z-index: 2;
}