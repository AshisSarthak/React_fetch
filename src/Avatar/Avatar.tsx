import React from "react";
import "./Avatar.scss";

export const Avatar = (props: { name: string; selectUser: () => void }) => {
  const splittedName: string[] = props.name.split(" ");
  const avatarText = splittedName[0].charAt(0) + splittedName[1].charAt(0);
  return (
    <div className="avatar" onClick={props.selectUser}>
      {avatarText}
    </div>
  );
};
