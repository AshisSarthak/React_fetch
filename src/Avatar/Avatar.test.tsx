import React from "react";
import { shallow } from "enzyme";
import { Avatar } from "./Avatar";

describe("Avatar Componnet", () => {
  it("default render", () => {
    const selectUserFunction = jest.fn();
    const wrapper = shallow(
      <Avatar name={"Ashis Sarthak"} selectUser={selectUserFunction} />
    );
    expect(wrapper).toMatchSnapshot();
    const innerDiv = wrapper.find("div").at(0);
    expect(innerDiv.text()).toBe("AS");
    innerDiv.simulate("click");
    expect(selectUserFunction).toBeCalled();
  });
});
