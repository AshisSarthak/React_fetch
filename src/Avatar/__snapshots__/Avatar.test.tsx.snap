// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Componnet default render 1`] = `
<div
  className="avatar"
  onClick={[MockFunction]}
>
  AS
</div>
`;
